full/src/main/java/apoc/custom/CypherProceduresHandler.java

```diff
@@ -98,8 +98,8 @@
     private final JobScheduler jobScheduler;
     private long lastUpdate;
     private final ThrowingFunction<Context, Transaction, ProcedureException> transactionComponentFunction;
-    private Set<ProcedureSignature> registeredProcedureSignatures = emptySet();
-    private Set<UserFunctionSignature> registeredUserFunctionSignatures = emptySet();
+    private final Set<ProcedureSignature> registeredProcedureSignatures = Collections.synchronizedSet(new HashSet<>());
```

Comment:
how does this get populated when the DB starts?

Reply:
In `available()` method, called after each reboot, through the `restoreProceduresAndFunctions()`, where it is populated based on System nodes.